app/src/test/java/io/gnosis/safe/ui/transactions/details/ButtonStateHelperTest.kt

```diff
@@ -6,299 +6,213 @@ import org.junit.Test
 
 class ButtonStateHelperTest {
 
-    // Rejections For numbering, see: https://docs.google.com/document/d/1dlhVAv9WiD1EPi1LuZI6opk17ivD2lpMbwbWP4aArIY/edit
+    // For numbering, see: https://docs.google.com/document/d/13GirjSwieQaqUPz-8HXzJP_rfgmobSp7E4TpbnLlb0M/edit#
     @Test
-    fun `(1) when (isRejection, !needsYourConfirmation, hasBeenRejected, !needsExecution) should hide all buttons`() {
+    fun `(1_1) when (isRejection, awaitingConfirmations, canSign) should hide rejection and enable confirmation button`() {
+        val awaitingConfirmations = true
         val buttonStateHelper =
             ButtonStateHelper(
                 isRejection = true,
-                needsYourConfirmation = false,
+                awaitingConfirmations = awaitingConfirmations,
+                hasOwnerKey = true,
+                canSign = true,
                 hasBeenRejected = true,
-                needsExecution = false,
-                canReject = true,
-                isOwner = true,
-                completed = false
+                awaitingExecution = false
             )
 
-        assertFalse(buttonStateHelper.confirmButtonIsVisible())
-        assertFalse(buttonStateHelper.rejectButtonIsVisible())
-        assertFalse(buttonStateHelper.buttonContainerIsVisible())
+        assertTrue(buttonStateHelper.confirmationButtonIsVisible())
+        assertTrue(buttonStateHelper.confirmationButtonIsEnabled())
+        assertTrue(buttonStateHelper.buttonContainerIsVisible())
```

Comment:
How come we asserted for the direct opposite before? The button container visibility should not have changed? Although asserting to `true` makes more sense than `false`.

Reply:
There is no relation between the old test class and the new test class. I think this being opposite is just a coincidence.